### PR TITLE
transform_function's first draft

### LINKED ISSUE #26: Create 3 simple testing dataframes
- Have df's set up so we can set up test cases first

### LINKED ISSUE #33: Consistency in naming conventions for dictionary keys
We'll need to ensure that all our functions use same name for the keys of the dictionary  returned by @cgostic 's function so we can fetch relevant column names by using those keys.

We may create our own keys as of now but eventually we'll need to have consistency in terms of naming conventions for the keys which all our functions will use!

### LINKED ISSUE #38: Create column_transform() function
Create column trasnform function to do transformations to numeric and categorical columns of dataset(test/train)

### PR DESCRIPTION
**Pull request by arunmarria**
The function is in its initial stage, I will keep on updating this PR by making necessary changes.

### CODE REVIEW

File: pylaundry/transform_columns.py [hunk 1]
```diff
@@ -18,8 +25,108 @@ def transform_columns(X, column_dict, cat_trans, num_trans):
...
+        # ## Applying transformations to training data set
+        X_train = pd.DataFrame(preprocessor.fit_transform(X_train), 
+                                   index = X_train.index,
+                                   columns = numeric +list(
+                                           preprocessor.named_transformers_['ohe'].get_feature_names(numeric)))
+            
+        #applying transformations to test set
+        X_test = pd.DataFrame(preprocessor.transform(X_test),
+                              index = X_test.index,
+                              columns= X_train.columns)
```
Comment (amank90) [inline pylaundry/transform_columns.py]:
```suggestion
Can change the X_train and X_test to X_train_transform and X_test_transform respectively
```

File: pylaundry/transform_columns.py [hunk 2]
```diff
@@ -18,8 +25,101 @@ def transform_columns(X, column_dict, cat_trans, num_trans):
...
+        
+        # ## Applying transformations to training data set
+        X_train = pd.DataFrame(preprocessor.fit_transform(X_train), 
+                                   index = X_train.index,
+                                   columns = numeric +list(
+                                           preprocessor.named_transformers_['ohe'].get_feature_names(numeric)))
+            
+        #applying transformations to test set
+        X_test = pd.DataFrame(preprocessor.transform(X_test),
+                              index = X_test.index,
```
Comment (amank90) [inline pylaundry/transform_columns.py]:
```suggestion

#Change X_test and X_train
```

File: pylaundry/transform_columns.py [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
     Transforms categorical and numerical features based on user input.
 
     Arguments
     ---------     
-    df -- pandas.core.frame.DataFrame       
-        A pandas dataframe
+    X_train -- pandas.core.frame.DataFrame       
+        A pandas dataframe for training set
+    X_test -- pandas.core.frame.DataFrame       
+        A pandas dataframe for training set
```
Comment (zanderhinton) [inline pylaundry/transform_columns.py]:
```suggestion
        A pandas dataframe for test set
```

File: pylaundry/transform_columns.py [hunk 4]
```diff
@@ -18,8 +25,101 @@ def transform_columns(X, column_dict, cat_trans, num_trans):
...
+    transformed_dict -- dictionary 
+        A python dictionary with transformed training and test set with keys X_train and X_test respectively
     
     """
-    pass
\ No newline at end of file
+    
+    ## code to retain name convention for function arguments
+    X_train = X_train
+    X_test = X_test
```
Comment (zanderhinton) [inline pylaundry/transform_columns.py]:
```suggestion
```

File: pylaundry/transform_columns.py [hunk 5]
```diff
@@ -18,8 +25,101 @@ def transform_columns(X, column_dict, cat_trans, num_trans):
...
+    
+    ## checking user inputs
+    
+     # assertions for test and train set inputs
+    assert isinstance(X_train, pd.DataFrame), "X_train should be a DataFrame"
+    assert isinstance(X_test, pd.DataFrame), "X_test should be a data frame"
```
Comment (zanderhinton) [inline pylaundry/transform_columns.py]:
```suggestion
    assert isinstance(X_test, pd.DataFrame), "X_test should be a DataFrame"
```

File: pylaundry/transform_columns.py [hunk 6]
```diff
@@ -18,8 +25,101 @@ def transform_columns(X, column_dict, cat_trans, num_trans):
...
+    assert [key in [ 'categorical', 'numeric'] for key in column_dict.keys() ], "column_dict keys can be only 'numeric' and 'categorical'"
+    
+    
+    # assertions for transformation inputs
+    assert isinstance(num_trans, str), "num_trans should be a string"
+    assert isinstance(cat_trans, str), "cat_trans should be a string"
+    assert num_trans == "standard_scaling" or num_trans == "minmax_scaling","transformation method for numeric columns can only be 'minmax_scaling' or 'standard_scaling'"
+    assert cat_trans == "onehot_encoding" or cat_trans == "label_encoding","transformation method for categorical columns can only be 'label_encoding' or 'onehot_encoding'"
+    
+
```
Comment (zanderhinton) [inline pylaundry/transform_columns.py]:
```suggestion
    # Check train set and test set columns are the same
    assert np.array_equal(train_df.columns, test_df.columns), "train_df and test_df must have the same columns"
```

**Review by zanderhinton — CHANGES_REQUESTED** [review]
Function looks very good, and is written very clean and easy to read, great! 
I have suggested a few minor editing changes, as well as adding in one additional edge case check to confirm columns are the same for train and test sets.

**Review by zanderhinton — APPROVED** [review]
New updates based on code review look  good!
